pymtl3/dsl/Component.py

```diff
@@ -45,17 +45,19 @@ def _construct( s ):
 
       s._handle_decorated_methods()
 
-      # We hook up the added clk and reset signals here.
+      # Same as parent class _construct
+      s.construct( *s._dsl.args, **kwargs )
+
+      # We hook up the added clk and reset signals here. NOTE THAT if the
+      # user overwrites clk/reset inside the component, we still get the
+      # correct connection.
       try:
         parent = s.get_parent_object()
         parent.connect( s.clk, parent.clk )
```

Comment:
is it possible that some statements inside `s.connect` throws an attribute error?

Reply:
I had some situations where I use try-except to catch some attribute error but unintentionally missed exceptions thrown from the function calls inside the block.

Reply:
oh good catch! Do I actually need this try-catch block here? I think the reason I had this was when I instantiate a ComponentLevel6 as a child model inside ComponentLevel3 and ComponengLevel3 doesn't have reset/clk. Now since we are exposing only Component to the user, I guess we don't need this anymore.